### DISCUSSION TITLE
Way to add postinstallscript option while creating First-Generation managed package version using SFDX

### DISCUSSION THREAD

**Discussion by yogeshdixit41** [forcedotcom/cli#2263] [Ideas]
**Is your feature request related to a problem? Please describe.**
I am trying to create a managed package version of my app using a set of SFDX commands to deploy the metadata and then build beta and prod packages (as per the document here https://developer.salesforce.com/docs/atlas.en-us.sfdx_dev.meta/sfdx_dev/sfdx_dev_build_man_pack_create.htm). I want to add the post-install script value to the apckage that gets created as a result but every time I build the package using sfdx commands it fails to add the post-install value to the package.

**What are you trying to do**
The package that gets created should have the post-install script value on it. I have also tried adding the post-install script value on the package and then using the `sfdx force:package1:version:create` to create a new version of the package in the hope of retaining the post-install script value but everytime the new version is created the post-install script value is erased.

**Describe the solution you'd like**
`sfdx force:package1:version:create` command should provide an option (--postinstallscript) to add the post-install value or support the postInstallScript value from the sfdx-project.json. 2GP package creation sfdx command (`sfdx force:package:version:create`) already supports both these options.

**Describe alternatives you've considered**
Unfortunately, the only alternative, in this case, is to manually create the package from packaging org which defeats the purpose of using sfdx commands and building the CI/CD process for maintaining the managed package app project

**Additional context**
https://developer.salesforce.com/docs/atlas.en-us.sfdx_dev.meta/sfdx_dev/sfdx_dev_dev2gp_customize_installs.htm
Here is the document that shows the support for the postinstallscript option for the 2nd Generation package creation command

[a comment by an automated account is not shown]

**Comment by yogeshdixit41** [comment]
Any update on this? Is there going to be any support added for the post-installation script option for the 1GP packaging commands? 
@WillieRuemmele I see you converted this one from issue to discussion, any specific reason why?

**Comment by mshanemc** [comment]
We're not going to invest in features for 1GP because of Packaging's roadmap for 2GP transition.  2GP via `package version create` already has `--post-install-script`

**Reply by yogeshdixit41** [reply]
thanks for the update @mshanemc , is it now possible to migrate the 1 GP managed package to 2 GP managed package? As per the document here (https://developer.salesforce.com/docs/atlas.en-us.pkg2_dev.meta/pkg2_dev/sfdx_dev_dev2gp.htm), it says the migration is not possible You can’t currently migrate a first-generation managed package to a second-generation managed package but this document states otherwise https://developer.salesforce.com/blogs/2023/05/move-to-managed-2gp-with-package-migrations

![Screenshot2024_01_23_104050](https://github.com/forcedotcom/cli/assets/6316101/9f824a0a-09cd-435e-aeea-4e9db41248d7)

**Reply by shetzel** [reply]
@yogeshdixit41 
>I see you converted this one from issue to discussion, any specific reason why?

We converted issues that are feature requests to discussions for tracking purposes and to promote discussion on the feature.

This section from the 2nd doc you linked states:
>To test out Package Migrations, you need to be an ISV partner with access to the [Salesforce Partner Community](https://partners.salesforce.com/)....On this channel, you will find a number of useful links, including a form to register for the Developer Preview. Once you’re activated, you can start testing Package Migrations."

So if that fits your situation then you can sign up for the Dev Preview and convert.
